src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationPool.cs

```diff
@@ -108,7 +112,21 @@ private void OnBlockAdded(object? sender, BlockReplacementEventArgs e)
         {
             try
             {
-                _headBlocksChannel.Writer.TryWrite(e);
+                _headBlocksReplacementChannel.Writer.TryWrite(e);
+            }
+            catch (Exception exception)
+            {
+                if (_logger.IsError)
```

Comment:
1. What exceptions are you expecting here?
2. TryWrite probably returns a bool

Reply:
i am not sure, i copied @marcindsobczak 's code from above but modified?
@marcindsobczak thoughts?

Reply:
We have the same in TxPool, it evolved through several changes to this form. try-catch was there before changing old method to TryWrite. I don't remember what kind of exceptions we faced when changed it to this form during TxPool crisis in August 2021. Maybe now we can do it better, but I don't know if it makes sense to change it when it works properly

Reply:
ok